pkg/controller/rpaasinstance/rpaasinstance_controller.go

```diff
@@ -404,6 +426,99 @@ func (r *ReconcileRpaasInstance) reconcileNginx(nginx *nginxv1alpha1.Nginx) erro
 	return err
 }
 
+func (r *ReconcileRpaasInstance) reconcileCacheHeaterVolume(instance *v1alpha1.RpaasInstance, plan *v1alpha1.RpaasPlan) error {
+	pvcName := instance.Name + "-heater-volume"
+
+	pvc := &corev1.PersistentVolumeClaim{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: pvcName, Namespace: instance.Namespace}, pvc)
+	isNotFound := k8sErrors.IsNotFound(err)
+	if err != nil && !isNotFound {
+		return err
+	} else if !isNotFound {
+		logrus.Infof("PersistentVolumeClaim %s is found, skipping creation", pvcName)
+		return nil
+	}
+
+	cacheHeaterStorage := &v1alpha1.CacheHeaterStorage{}
+	if plan.Spec.Config.CacheHeaterStorage != nil {
+		cacheHeaterStorage = plan.Spec.Config.CacheHeaterStorage
+	}
+
+	volumeMode := corev1.PersistentVolumeFilesystem
+  labels := map[string]string{}
+  if cacheHeaterStorage.VolumeLabels != nil {
+    labels = cacheHeaterStorage.VolumeLabels
```

Comment:
@bernardolins you need to use `for` to copy a map[string]string

Reply:
Done